src/tools/debugger/frontend/Suspension.java

```diff
@@ -145,12 +148,21 @@ public void suspend() {
     while (continueWaiting) {
       try {
         continueWaiting = tasks.take().execute();
-        SteppingStrategy strategy = activityThread.getSteppingStrategy();
-        if (!continueWaiting && strategy != null) {
-          strategy.handleResumeExecution(activity);
-          strategy.handleMessageToNextTurn(activity);
-        }
 
+        if (!continueWaiting) {
+          if (activityThread.isStepping(SteppingType.RETURN_FROM_ACTIVITY)) {
+            activity.setStepToJoin(true);
+          } else if (activityThread.isStepping(SteppingType.STEP_TO_NEXT_TURN)) {
+            activity.setStepToNextTurn(true);
+          } else if (activityThread.isStepping(SteppingType.RETURN_FROM_TURN_TO_PROMISE_RESOLUTION)) {
```

Comment:
I think I understand what you mean here, is similar to the step-to-next-turn command in the sense that should be able to trigger from anywhere in the code. And I only have it implemented for the `SendNode` and the primitives of `PromisePrims`